src/launcher.c

```diff
@@ -171,22 +190,46 @@ int replace_current_code(struct app_s *app)
     return -1;
   }
 
-  uint32_t start = 0;
-  uint32_t code_length = app->elf.load_size;
-  if (app->elf.code_start != 0 && app->elf.code_end != 0) {
-    start = app->elf.code_start - TEXT_START;
-    code_length = app->elf.code_end - app->elf.code_start;
-  }
-
   if (mprotect(memory.code, app->elf.load_size, PROT_READ | PROT_WRITE) != 0) {
     warn("could not update mprotect in rw mode for app");
     _exit(1);
   }
 
-  if (patch_svc(memory.code + start, code_length) != 0) {
-    /* this should never happen, because the svc were already patched without
-     * error during the first load */
-    _exit(1);
+  uint32_t text_start = ((hw_model == MODEL_NANO_S) || (hw_model == MODEL_BLUE))
+                            ? LNS_TEXT_START
+                            : TEXT_START;
+
+  // If svc_call_addr or svc_cx_call_addr are defined (non 0),
+  // patch the elf at this address to remove the SYSCALL
+  if (!is_svc_inlined() &&
+      (app->elf.svc_call_addr != 0 || app->elf.svc_cx_call_addr != 0)) {
+    if (app->elf.svc_call_addr != 0) {
+      uint32_t start = app->elf.svc_call_addr - text_start;
+      uint32_t code_length = 2;
+
+      if (patch_svc(memory.code + start, code_length) != 0) {
+        /* this should never happen, because the svc were already patched
+         * without error during the first load */
+        _exit(1);
+      }
+    }
```

Comment:
This block appears a lot, can you make a function with it?

Reply:
+1